schema/mysql/upgrades/1.0.0-rc2.sql

```diff
@@ -141,3 +141,14 @@ ALTER TABLE service_state
     MODIFY output mediumtext DEFAULT NULL,
     MODIFY long_output mediumtext DEFAULT NULL,
     MODIFY performance_data mediumtext DEFAULT NULL;
+
+ALTER TABLE history
+    ADD CONSTRAINT fk_history_acknowledgement_history FOREIGN KEY (acknowledgement_history_id) REFERENCES acknowledgement_history (id) ON DELETE CASCADE,
+    ADD CONSTRAINT fk_history_comment_history FOREIGN KEY (comment_history_id) REFERENCES comment_history (comment_id) ON DELETE CASCADE,
+    ADD CONSTRAINT fk_history_downtime_history FOREIGN KEY (downtime_history_id) REFERENCES downtime_history (downtime_id) ON DELETE CASCADE,
+    ADD CONSTRAINT fk_history_flapping_history FOREIGN KEY (flapping_history_id) REFERENCES flapping_history (id) ON DELETE CASCADE,
+    ADD CONSTRAINT fk_history_notification_history FOREIGN KEY (notification_history_id) REFERENCES notification_history (id) ON DELETE CASCADE,
+    ADD CONSTRAINT fk_history_state_history FOREIGN KEY (state_history_id) REFERENCES state_history (id) ON DELETE CASCADE;
```

Comment:
What's the exact point of ON DELETE CASCADE here?

Reply:
For details on that commit, you'd have to ask @raviks789, that one is just taken from #259. But from my understanding, it's for #249. That's the reason why we started considering foreign key constraints in the first place.

Reply:
OK, let's assume your guess is right. IYHO: how **exactly** are these FKs supposed to work for history cleanup?

Reply:
Events are deleted from the details table and should delete the corresponding entries in the `history` table with them (from #249): https://github.com/Icinga/icingadb/blob/201197b187e05beef5bc93a719caa882bfdd752d/pkg/cleanup/cleanup.go#L143-L148

Reply:
Wouldn't it be smarter to reverse that rel-ship to only have to delete from one table and everything else cascades?

Reply:
The queries I linked only delete related events (i.e. downtime start and end) after both are beyond the deletion threshold. Also not sure how you'd implement this in detail as there's not a 1:1 relation but a 1:N (with N limited to a maximum of 2 though I guess).

Reply:
> delete related events (i.e. downtime start and end) after both are beyond the deletion threshold

OK, then we can't do this:

> delete from one table and everything else cascades

Reply:
In which scenario? What the PR currently does or what you suggested?

Reply:
I just threw my idea (second quote) away as the more important feature (first quote) blocks it. Nothing to worry about.